apps/meteor/app/api/server/router.ts

```diff
@@ -19,214 +14,38 @@ function splitArray<T, U>(arr: [...T[], U]): [T[], U] {
-						status,
-						{
-							description: '',
-							content: {
-								'application/json': 'schema' in schema ? { schema: schema.schema } : schema,
-							},
-						},
-					]),
-				),
-			}),
-			...(query && {
-				parameters: [
-					{
-						schema: query.schema,
-						in: 'query',
-						name: 'query',
-						required: true,
-					},
-				],
-			}),
-			...(body && {
-				requestBody: {
-					required: true,
-					content: {
-						'application/json': { schema: body.schema },
-					},
-				},
-			}),
-			...(authRequired && {
-				...rest,
-				security: [
-					{
-						userId: [],
-						authToken: [],
-					},
-				],
-			}),
-			tags,
-		};
-	}
-
-	private async parseBodyParams(request: HonoRequest, overrideBodyParams: Record<string, any> = {}) {
-		if (Object.keys(overrideBodyParams).length !== 0) {
-			return overrideBodyParams;
+> extends Router<TBasePath, TOperations> {
+	protected async parseBodyParams<T extends Record<string, any>>({ request, extra }: { request: HonoRequest; extra?: T }) {
+		if (extra && Object.keys(extra.bodyParamsOverride || {}).length !== 0) {
+			return extra;
 		}
 
-		try {
-			let parsedBody = {};
-			const contentType = request.header('content-type');
-
-			if (contentType?.includes('application/json')) {
-				parsedBody = await request.raw.clone().json();
-			} else if (contentType?.includes('multipart/form-data')) {
-				parsedBody = await request.raw.clone().formData();
-			} else if (contentType?.includes('application/x-www-form-urlencoded')) {
-				const req = await request.raw.clone().formData();
-				parsedBody = Object.fromEntries(req.entries());
-			} else {
-				parsedBody = await request.raw.clone().text();
-			}
-			// This is necessary to keep the compatibility with the previous version, otherwise the bodyParams will be an empty string when no content-type is sent
-			if (parsedBody === '') {
-				return {};
-			}
-
-			if (Array.isArray(parsedBody)) {
-				return parsedBody;
-			}
-
-			return { ...parsedBody };
-			// eslint-disable-next-line no-empty
-		} catch {}
-
-		return {};
+		return super.parseBodyParams({ request });
 	}
 
-	private parseQueryParams(request: HonoRequest) {
-		return qs.parse(request.raw.url.split('?')?.[1] || '');
-	}
-
-	private method<TSubPathPattern extends string, TOptions extends TypedOptions, TPathPattern extends `${TBasePath}/${TSubPathPattern}`>(
-		method: Method,
-		subpath: TSubPathPattern,
-		options: TOptions,
-		...actions: MiddlewareHandlerListAndActionHandler<TOptions, TSubPathPattern>
-	): Router<
-		TBasePath,
-		| TOperations
-		| ({
-				method: Method;
-				path: TPathPattern;
-		  } & Omit<TOptions, 'response'>)
-	> {
-		const [middlewares, action] = splitArray(actions);
-
-		this.innerRouter[method.toLowerCase() as Lowercase<Method>](`/${subpath}`.replace('//', '/'), ...middlewares, async (c) => {
+	protected _convertToHonoAction(action: TypedAction<any, any>): (c: HonoContext) => Promise<any> {
```

Comment:
move the whole class to the external package, only implements this method (maybe find a more generic name), also, make the context generic on the interface context